admin/services/generate-pins.service.js

```diff
@@ -63,14 +79,48 @@ const generatePinService = {
     return school
   },
 
+  /**
+   * Validate pin
+   * @param pin
+   * @param pinExpiresAt
+   * @returns {Boolean}
+   */
   isValidPin: (pin, pinExpiresAt) => {
     if (!pinExpiresAt || !pin) return false
     return moment(pinExpiresAt).isAfter(moment.utc())
   },
 
+  /**
+   * Generate Random Pin
+   * @param length
+   * @returns {String}
+   */
   generateRandomPin: (length) => {
     const chars = '23456789bcdfghjkmnpqrstvwxyz'
     return randomGenerator.getRandom(length, chars)
+  },
+
+  /**
+   * Get active school Password
+   * @param school
+   * @returns {String}
+   */
+  getActiveSchool: async (schoolId) => {
+    const school = await schoolDataService.findOne({_id: schoolId})
+    if (generatePinService.isValidPin(school.schoolPin, school.pinExpiresAt)) {
+      return school
+    } else return null
+  },
+
+  /**
+   * Get pupils with active pins
+   * @param schoolId
+   * @returns {Array}
+   */
+  getPupilsWithActivePins: async (schoolId) => {
+    let pupils = await pupilDataService.getSortedPupils(schoolId, 'lastName', 'asc')
+    pupils = pupils.filter(p => generatePinService.isValidPin(p.pin, p.pinExpiresAt))
```

Comment:
I would have thought that the DB could applied the filter

Reply:
Hmmm I don't mind really but I do have the isValidPIn common service method that makes it easy enough to filter.
If I'd have to do it on mongoose level basically I would be duplicating this logic instead of reusing it.